File: parfumerie-frontend/src/app/services/employe.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { EmployeDTO } from '../models/employe.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class EmployeService {

  private baseUrl = 'http://localhost:8080/employe';

  constructor(private http: HttpClient) { }

  creer(employe: EmployeDTO): Observable<EmployeDTO> {
    return this.http.post<EmployeDTO>(`${this.baseUrl}/create`, employe);
  }

  modifier(id: number, employe: EmployeDTO): Observable<EmployeDTO> {
    return this.http.put<EmployeDTO>(`${this.baseUrl}/update/${id}`, employe);
  }

  consulterById(id: number): Observable<EmployeDTO> {
    return this.http.get<EmployeDTO>(`${this.baseUrl}/consult/${id}`);
  }

  consulter(): Observable<EmployeDTO[]> {
    return this.http.get<EmployeDTO[]>(`${this.baseUrl}/consult`);
  }

  supprimer(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/delete/${id}`);
  }
  
}

File: parfumerie-frontend/parfumerie-backend/src/main/java/com/parfumerie/gestion_parfumerie/services/EmployeService.java
package com.parfumerie.gestion_parfumerie.services;

import java.util.List;

import com.parfumerie.gestion_parfumerie.dto.EmployeDTO;

public interface EmployeService {
	
	
	void supprimer(Long id);
	EmployeDTO consulterById(Long id);
	List<EmployeDTO> consulter();
	EmployeDTO modifier(Long id, EmployeDTO dto);
	EmployeDTO creer(EmployeDTO dto);
	
}
